main.tf

```
@@ -1,18 +1,27 @@
 resource "aws_cloudwatch_event_rule" "this" {
 
-  name          = var.event_rule.name
-  description   = var.event_rule.description
-  event_pattern = var.event_rule.event_pattern
+  name           = var.event_rule.name
+  description    = var.event_rule.description
+  event_pattern  = var.event_rule.event_pattern
+  event_bus_name = var.event_rule.event_bus_name
 }
 
 resource "aws_cloudwatch_event_target" "this" {
 
   for_each = { for target in var.event_rule.event_targets : target.name => target }
 
-  arn       = each.value.arn
-  rule      = aws_cloudwatch_event_rule.this.id
-  target_id = each.value.target_id
-  role_arn  = each.value.role_arn
+  arn            = each.value.arn
+  rule           = aws_cloudwatch_event_rule.this.id
+  target_id      = each.value.target_id
+  role_arn       = each.value.role_arn
+  event_bus_name = each.value.event_bus_name
+
+  dynamic "dead_letter_config" {
+    for_each = each.value.dead_letter_config != null ? [1] : []
+    content {
+      arn = each.value.dead_letter_config.arn
+    }
+  }
```

use the dynamic block as the iteration object

```suggestion
  dynamic "dead_letter_config" {
    for_each = each.value.dead_letter_config != null ? [each.value.dead_letter_config] : []
    content {
      arn = dead_letter_config.value.arn
    }
  }
```

---

yeah, i had written it this way, but terraform refuses to recognize the object attribute in the content {} section